float32 acc
---
bool succeded
string error_msg
